javascript/ql/lib/semmle/javascript/frameworks/Express.qll

```diff
@@ -901,9 +900,9 @@ module Express {
 
     /**
      * DEPRECATED: Use `ref().flowsToExpr()` instead.
-     * Holds if `sink` may refer to this router.
+     * This predicate is now from `DataFlow::SourceNode`.
      */
-    deprecated predicate flowsTo(Expr sink) { this.ref().flowsToExpr(sink) }
+    deprecated override predicate flowsTo(DataFlow::Node n) { super.flowsTo(n) }
```

Comment:
It can be removed, but if you do that then previous uses of `RouterDefinition::flowsTo` would suddenly point to `SourceNode::flowsTo` without a deprecation warning, which would be a major change in behavior.  

That gotcha happened to me during this rewrite, and it caused weird behavior changes that took me a while to debug.   
So I prefer keeping this deprecation.